### DISCUSSION TITLE
How to convert a Godot Dictionary to and from a Go (Golang) struct

### DISCUSSION THREAD

**Discussion by KKKFFFFFF** [quaadgras/graphics.gd#59] [Q&A]
How to convert a Godot Dictionary to and from a Go (Golang) struct, and whether it can be converted into a Protobuf structure.Are there any places for asking questions and exchanging ideas?

**Comment by Splizard** [comment]
I've converted this into a discussion, check out the other discussions for a place to ask questions and share ideas:
https://github.com/grow-graphics/gd/discussions

Thanks for sharing this idea though, I think it could make sense to officially support this and to include ways to convert structs into dictionaries. This can be achieved through the use of reflection, although bear in mind that there would be an allocation overhead to doing this.

**Reply by KKKFFFFFF** [reply]
case data := <-k.pbChan:
			router := data.GetRouter()
			fmt.Println(router, "   router")
			tmp := k.Temporary
			//dataMap := gd.NewMap[gd.String, any](tmp)
			switch router {
			case 1:
				fmt.Println("PackDictionaryToGD with ID :", router)
				l := gd.NewLifetime(tmp)
				dataMap := gd.NewMap[gd.String, any](l)
				dataMap.Set(l.String("router"), gd.Int(router))
				dataMap.Set(l.String("message"), l.String(data.GetText()))
				k.MapRequest.Emit(gd.Dictionary(dataMap))
				l.Free()
			case 2:
				fmt.Println("PackDictionaryToGD with ID :", router)
				l := gd.NewLifetime(tmp)
				dataMap := gd.NewMap[gd.String, any](l)
				dataMap.Set(l.String("router"), gd.Int(router))
				dataMap.Set(l.String("message"), l.String(data.GetText()))
				k.PlayerMgrRequest.Emit(gd.Dictionary(dataMap))
				l.Free()
			case 3:
				fmt.Println("PackDictionaryToGD with ID :", router)
				l := gd.NewLifetime(tmp)
				dataMap := gd.NewMap[gd.String, any](l)
				dataMap.Set(l.String("router"), gd.Int(router))
				dataMap.Set(l.String("message"), l.String(data.GetText()))
				k.ChatRequest.Emit(gd.Dictionary(dataMap))
				l.Free()
"I am currently using this method to convert struct data into a dictionary, and I find it too cumbersome."

**Reply by KKKFFFFFF** [reply]
"Additionally, I hope to have a method that can convert a dictionary into a Protobuf type."
